turbopack/crates/turbo-tasks/src/capture_future.rs

```diff
@@ -80,19 +100,46 @@ impl<T, F: Future<Output = T>> Future for CaptureFuture<T, F> {
         EXTRA.with_borrow_mut(|cell| {
             *cell = Some(&mut data as *mut ThreadLocalData);
         });
-        let result = this.future.poll(cx);
+
+        let result =
+            panic::catch_unwind(panic::AssertUnwindSafe(|| this.future.poll(cx))).map_err(|err| {
+                let message = match err.downcast_ref::<&'static str>() {
+                    Some(s) => TurboTasksExecutionErrorMessage::PIISafe(s),
+                    None => match err.downcast_ref::<String>() {
+                        Some(s) => TurboTasksExecutionErrorMessage::NonPIISafe(s.clone()),
+                        None => {
+                            let error_message = err
+                                .downcast_ref::<Box<dyn Error + 'static>>()
```

Comment:
Does this work?

```suggestion
                                .downcast_ref::<dyn Display>()
```

I think calling `downcast_ref::<Box<_>>` is going to expect that the error is `Box<Box<_>>`, which isn't what you want. If you really needed an owned `Box` for some reason, you'd need to use `Any::downcast` instead of `Any::downcast_ref`: https://doc.rust-lang.org/std/any/trait.Any.html#method.downcast

You don't really need an `Error`... It seems like you just need something you can call `to_string()` on.